relay-server/src/actors/processor.rs

```diff
@@ -1124,6 +1125,50 @@ impl EnvelopeProcessorService {
         });
     }
 
+    fn process_recording_timed(
+        &self,
+        bytes: &[u8],
+        limit: usize,
+    ) -> Result<Vec<u8>, recording::RecordingParseError> {
+        // Check for null byte condition.
+        if bytes.is_empty() {
+            return Err(recording::RecordingParseError::Message("no data found"));
+        }
+
+        let mut split = bytes.splitn(2, |b| b == &b'\n');
+        let header = split
+            .next()
+            .ok_or(recording::RecordingParseError::Message("no headers found"))?;
+
+        let body = match split.next() {
+            Some(b"") | None => {
+                return Err(recording::RecordingParseError::Message("no body found"))
+            }
+            Some(body) => body,
+        };
+
+        let buffer = metric!(timer(RelayTimers::ReplayRecordingDecompress), {
+            recording::decompress(body, limit)?
+        });
+        let mut events = metric!(timer(RelayTimers::ReplayRecordingDeserialize), {
+            recording::deserialize(buffer)?
+        });
+
+        metric!(timer(RelayTimers::ReplayRecordingScrubPII), {
+            recording::strip_pii(&mut events)
+                .map_err(recording::RecordingParseError::ProcessingAction)?
+        });
+
+        let bytes = metric!(timer(RelayTimers::ReplayRecordingSerialize), {
+            recording::serialize(events)?
+        });
+        let compressed_bytes = metric!(timer(RelayTimers::ReplayRecordingCompress), {
+            recording::compress(bytes)?
+        });
```

Comment:
Should we also add histogram metrics for the original compressed and decompressed size, + processed compressed and decompressed size?

Usage example: https://github.com/getsentry/relay/blob/1bf9c64d85a1a3d91c57d6f9fab77aed551e904d/relay-server/src/body/store_body.rs#L46